File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ## edge cases
        if len(nums) == 0:
            return

        ## calculate the formula
        ## nums[i] = target - nums[j]
        ## check if nums[i] is present in target - nums[j]
        pMap = {}
        for i, n in enumerate(nums):
            findJ = target - n

            if findJ in pMap:
                return [pMap[findJ], i]
            else:
                pMap[n] = i


        ##return
        return

        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {

        // edge cases

        int l =0;
        int r = numbers.Length -1;

        while (l < r) {
            int sum = numbers[l] + numbers[r];
            if (sum < target) {
                l++;
            } else if (sum > target) {
                r--;
            }
            else if(sum == target) {
                return new int[] {l + 1, r + 1};
            }
        }

        return new int[] {-1, -1};
    }
}
